tooling/azure-automation/resources-cleanup/src/resources_cleanup.py

```diff
@@ -109,16 +108,26 @@ def process_resource_group(resource_client: ResourceManagementClient, resource_g
     now = datetime.datetime.now(datetime.timezone.utc)
     resource_group_creation_time = get_creation_time_of_resource_group(resource_group)
     if not time_delta_greater_than_two_days(now, resource_group_creation_time):
-        print(f"This resource group should NOT be deleted, skipping.")
+        print(f"This resource group should NOT be deleted, it is not older than two days, skipping.")
         return
     
-    print("This resource group should be deleted.\n")
-    if not DRY_RUN:
-        print("\nBeginning deletion of this resource group ...")
+    if DRY_RUN:
+        print("This resource group should be deleted.\n")
+        return
+    
+    try:
         result_poller = resource_client.resource_groups.begin_delete(resource_group_name)
+        print("This resource group should be deleted.\n")
```

Comment:
nit: You could move this message above `if DRY_RUN:` and eliminate the duplicate print statements.

Reply:
Done, thanks. 

I put it that way because:
- when in dry run mode, we can not check if the deny assignment will block the deletion as it happens when trying to delete it.
- we don't really want to show that message if the resource group has a deny assignment, and we know that when the line "result_poller = resource_client.resource_groups.begin_delete(resource_group_name)"raises an exception.

As it is right now, the script will possibly lie in dry-run mode when saying that the resource group should be deleted but the resource group has a deny assignment and then can not be deleted. When not in dry run mode it will not lie as the print will happen after the point where the exception can be raised.

But I am applying the suggestion to unify the expected output. 🙂